src/researches/getPassiveVoice.js

```diff
@@ -1,110 +1,122 @@
-var determinePassivesDefault = require( "./passiveVoice/determinePassives" );
+const forEach = require( "lodash/forEach" );
 
-// Imports used for German.
-var getSentencePartsGerman = require( "./german/passiveVoice/getSentenceParts.js" );
-var determinePassivesGerman = require( "./german/passiveVoice/determinePassives.js" );
+const determinePassiveSentencePart = require( "./passiveVoice/periphrastic/determinePassiveSentencePart.js" );
+const determinePassiveSentence = require( "./passiveVoice/morphological/determinePassiveSentence.js" );
 
-var forEach = require( "lodash/forEach" );
+const getSentencePartsDefault = require( "./passiveVoice/periphrastic/getSentenceParts.js" );
+const getSentencePartsGerman = require( "./german/passiveVoice/getSentenceParts.js" );
+
+const morphologicalLanguages = [ "ru", "tr" ];
+const periphrasticLanguages = [ "en", "de", "nl", "fr", "es", "it", "pt", "cn" ];
+const morphologicalAndPeriphrasticLanguages = [ "sv", "da", "nb" ];
 
 /**
- * Gets the sentence parts from a sentence by determining sentence breakers.
+ * Looks for morphological passive voice.
  *
- * @param {string} sentence The sentence to split up in sentence parts.
- * @param {string} language The language to use for determining how to get sentence parts.
- * @returns {Array} The array with all parts of a sentence that have an auxiliary.
+ * @param {Array} sentenceObjects Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {object} The number of passives found in the text and the passive sentences.
  */
-var getSentenceParts = function( sentence, language ) {
-	var sentenceParts = [];
-
-	switch( language ) {
-		case "de":
-			sentenceParts = getSentencePartsGerman( sentence );
-			break;
-		case "fr":
-			sentenceParts = getSentencePartsDefault( sentence, "fr" );
-			break;
-		case "es":
-			sentenceParts = getSentencePartsDefault( sentence, "es" );
-			break;
-		case "en":
-		default:
-			sentenceParts = getSentencePartsDefault( sentence, "en" );
-			break;
-	}
-	return sentenceParts;
+let getPassivesMorphological = function( sentenceObjects, language ) {
+	let passiveSentencesByType = [];
+
+	forEach( sentenceObjects, function( sentence ) {
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+
+		sentence.setPassive( determinePassiveSentence( strippedSentence, language ) );
+
+		if ( sentence.isPassive() === true ) {
+			passiveSentencesByType.push( sentence.getSentenceText() );
+		}
+	} );
+	return {
+		passiveSentencesByType,
+	};
 };
 
 /**
- * Checks the sentence part for any passive verb.
+ * Looks for periphrastic passive voice.
  *
- * @param {object} sentencePart The sentence part object to check for passives.
- * @param {string} language The language to use for finding passive verbs.
- * @returns {boolean} True if passive is found, false if no passive is found.
+ * @param {Array} sentenceObjects Sentences extracted from the text.
```

Comment:
Like above:

> The name of this can just be sentences. Try to avoid the type of variables in the name of variables. I know this hasn't been done consistently in the project, but that is legacy.

Reply:
Agreed, fixed.